### PR TITLE
[MM-26386] - Respect user locale when searching for channels

### PR DESCRIPTION
**Pull request by ghost**
https://mattermost.atlassian.net/browse/MM-26386

#### Summary
Respect the user locale when searching for channels instead of defaulting to English.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-26386

### CODE REVIEW

**Review by Willyfrog — APPROVED** [review]

**Comment by amyblais** [comment]
Pushing to v5.30 as this is not new and probably won't make it to today's code complete.

**Review by jgilliam17 — APPROVED** [review]
Thank you @nevyangelova 
Tested, looks good to merge.
- Verified channels switcher - works as expected. 
I'll add the 2 issues, about userid and help text, for follow up as we discussed.

File: components/suggestion/search_channel_provider.jsx [hunk 1]
```diff
@@ -51,10 +56,9 @@ export default class SearchChannelProvider extends Provider {
                         channels = channels.filter((ch) => isDirectChannel(ch) || isGroupChannel(ch));
                     }
 
-                    //
-                    // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
-                    //
-                    channels = channels.sort(sortChannelsByTypeListAndDisplayName.bind(null, 'en', [Constants.OPEN_CHANNEL, Constants.PRIVATE_CHANNEL, Constants.DM_CHANNEL, Constants.GM_CHANNEL]));
+                    const locale = getCurrentLocale(getState());
+
+                    channels = channels.sort(sortChannelsByTypeListAndDisplayName.bind(null, locale, [Constants.OPEN_CHANNEL, Constants.PRIVATE_CHANNEL, Constants.DM_CHANNEL, Constants.GM_CHANNEL]));
```
Comment (sudheerDev) [inline components/suggestion/search_channel_provider.jsx:61]: This looks fine but we are adding access to the store here, could we have avoided by passing them into the `handlePretextChanged`?

**Review by sudheerDev — APPROVED** [review]

**Pull request merged**